ranges.go

```diff
@@ -19,7 +19,10 @@ func (r Range) Empty() bool {
 
 // Length returns the number of elements in the range.
 func (r Range) Length() uint {
-	return r.End - r.Start
+	if r.End > r.Start {
+		return r.End - r.Start
+	}
+	return 0
```

Comment:
Good catch! But we should follow go convention & handle the exceptional case in the if block:

```suggestion
	if r.End < r.Start {
		return 0
	}
	return r.End - r.Start
```

Reply:
Flatter code is a general programming convention. I'm curious to know what makes you describe it as a Go convention, could you provide a link?

That said, it's your repo, and it's not like it makes any difference to me. I'll invert it.